src/redux/slices/user.ts

```diff
@@ -96,20 +101,29 @@ export const sendToken = createAsyncThunk(
 export const burnXtc = createAsyncThunk(
   'user/burnXtc',
   async (
-    params: { to: string; amount: string; subaccount: string },
+    params: {
+      to: string;
+      amount: string;
+      subaccount?: string;
+      onEnd?: () => void;
+    },
     { rejectWithValue }
   ) => {
     try {
+      const { to, amount, subaccount, onEnd } = params;
       const instance = KeyRing.getInstance();
-      const response = await instance?.burnXTC(params);
+      const response = await instance?.burnXTC({ to, amount, subaccount });
       if ('Ok' in response) {
+        onEnd?.();
```

Comment:
should we call it onFinish? or onEnd? We should choose one and refactor the others

Reply:
i prefer onSuccess and onFailure